rapids_build_backend/impls.py

```diff
@@ -167,11 +168,19 @@ def _edit_pyproject(config):
 
     cuda_version = _get_cuda_version(config.require_cuda)
 
+    # "dependencies.yaml" might not exist in sdists and wouldn't need to... so don't
+    # raise an exception if that file can't be found when this runs
```

Comment:
I think we should distinguish between "`dependencies.yaml` doesn't exist because it's an sdist" and "`dependencies.yaml` doesn't exist because the developer forgot to configure it properly." The former is easy to mark with some sort of marker file or a setting that gets written to `pyproject.toml`.

Reply:
> a setting that gets written to `pyproject.toml`

Maybe we should have this anyway, instead of assuming that the user doesn't want `dependencies.yaml` just because they forgot to include it.

Reply:
Other things about this project would need to change to support sdists too.

The relative paths that reach up above `pyproject.toml`, like this:

```toml
[tool.rapids-build-backend]
dependencies-file = "../../dependencies.yaml"
```

are unlikely to work in an `sdist`, where I'd expect `pyproject.toml` to be at the root of the file layout.

This is just another place that's exposing the tension between building from an sdist vs. the file layout in source control (https://github.com/rapidsai/rapids-build-backend/pull/17#discussion_r1538240715).

From my perspective, having `rapids-build-backend` be stricter here (raising an error if the file that `dependencies-file` points to does not exist) is useful at this stage of development, where we're only building wheels and don't know the exact shape of what building RAPIDS sdists would even look like (https://github.com/rapidsai/rapids-build-backend/pull/17#discussion_r1552372935).

But if we don't want to do that, then I think raising this warning is sufficient (to at least help with debugging mistakes like the one I made in `cudf`) and that we shouldn't try to add configuration to differentiate between the sdist vs misconfiguration distinction. I think any attempt to do that is going to be difficult to review without knowing what we want out of sdist support, and that sdist support isn't a high priority right now.

Reply:
I raised the sdist discussions, but I am also leaning towards us not needing to plan for them right now. I think it's causing additional complexity that isn't merited at this stage. I'd be fine with being stricter now and loosening up restrictions if and when we decide that we want to support sdists. The current state is a bit confusing because the backend does wrap the wrapped backend's `build_sdist` call correctly right now, which makes it seem like we support sdists, but I don't want to remove that stuff. For now we can leave it in whatever half-supported state is convenient.

Reply:
Ok thanks for that!

I think we should proceed right now with adding this *warning*, and defer any other changes related to this.

I spent an hour debugging issues in my `cudf` PR that had the root cause "misconfigured `dependencies-file`". Having this case be an error would have saved me 59 of those 60 minutes... having this result in a warning would have saved me 58 of those 60 minutes. So the warning gets most of the benefit I wanted out of this discussion, and leaves the library in a state where it's possible to experiment with sdists.

@KyleFromNVIDIA I just updated this to latest `main` now that you merged #30 and #32 . I think we should merge this and then cut a release.